components/brave_ads/core/internal/user_engagement/conversions/conversions_util.cc

```diff
@@ -36,8 +36,7 @@ bool CanConvertAdEvent(const AdEventInfo& ad_event) {
     }
 
     case AdType::kSearchResultAd: {
-      // Always.
-      return true;
+      return UserHasOptedInToSearchResultAds();
```

Comment:
@zenparsing this needs more thought, so I will discuss this with @aseren before we merge. Thanks

Reply:
This logic is good because the default value for the pref is true for non-rewards users and if a user resets rewards the pref is cleared to fallback to the default value.